common/src/main/java/com/graphicsfuzz/common/util/PipelineInfo.java

```diff
@@ -251,6 +251,35 @@ public void removeUniformBinding(String uniformName) {
     }
   }
 
+  public int appendValueToUniform(String uniformName, Number value) {
+
+    if (!dictionary.has(uniformName)) {
+      throw new IllegalArgumentException("Uniform declaration not found.");
+    }
+
+    dictionary.getAsJsonObject(uniformName).get("args").getAsJsonArray().add(value);
+
+    return dictionary.getAsJsonObject(uniformName).get("args").getAsJsonArray().size() - 1;
+  }
+
+  public int getUnusedBindingNumber() {
+    int number = 0;
+
+    List<Integer> bindings =
+        dictionary.entrySet().stream()
+            .filter(item -> item.getValue().getAsJsonObject().has("binding"))
+            .map(item -> item.getValue().getAsJsonObject()
+            .get("binding").getAsInt()).collect(Collectors.toList());
+
+    for (;number < Integer.MAX_VALUE; number++) {
+      if (bindings.contains(number)) {
+        break;
+      }
```

Comment:
This seems wrong - shouldn't you break if `bindings` does *not* contain `number`?

Also: instead of a break, it would be better to directly return `number`.

After the loop you could throw a new RuntimeException with a message saying that the code after the loop should be unreachable (as we should never have `MAX_VALUE` used bindings).

Reply:
Fixed the bug and changed as requested.